### PR TITLE
Added _source support

### PR DESCRIPTION
**Pull request by andybarilla**
I needed the ability to do source filtering as documented here: https://www.elastic.co/guide/en/elasticsearch/reference/1.7/search-request-source-filtering.html

Let me know if it needs tweaking. There's also a merge for master but I figured I'd make sure one of them is correct first.

### CODE REVIEW

File: elasticsearch_dsl/search.py [hunk 1]
```diff
@@ -382,6 +386,38 @@ def script_fields(self, **kwargs):
...
+            s = Search()
+            s = s.source("obj.*")
+
+            s = Search()
+            s = s.source(["obj1.*", "obj2.*"])
+
+            s = Search()
+            s = s.source({
+                "include": ["obj1.*", "obj2.*"],
+                "exclude": ["*.description"]
```
Comment (HonzaKral) [inline elasticsearch_dsl/search.py]: this seems cumbersome, how would you feel about using kwargs instead?
```
s = Search()
s = s.source(include=['obj1.*'], exclude=["*.description"])
```
with only the keyword specified being overridden, meaning the example could also be written as:
```
s = Search()
s = s.source(include=['obj1.*']).source(exclude=["*.description"])
```

**Comment by HonzaKral** [comment]
Thanks for the PR, it looks good and it is definitely useful functionality. I left a simple note on the diff.

Also, I would like to start deprecating the `1.x` branch, could you make the PR for master? Thanks!

**Comment by HonzaKral** [comment]
Closing in favor of #352

**Comment by levylll** [comment]
hi, i'd like to use the 'source' in my program, but I get the 'AttributeError: 'Search' object has no attribute 'source'' as my return.

**Comment by levylll** [comment]
@HonzaKral 
hi, i'd like to use the 'source' in my program, but I get the 'AttributeError: 'Search' object has no attribute 'source'' as my return.

**Pull request closed without merging**